src/AbstractOperations/binary_operations.jl

```diff
@@ -187,6 +193,40 @@ macro binary(ops...)
     return expr
 end
 
+#####
+##### BinaryOperations with ZeroField
+#####
+
+import Base: +, -, *, /, ==
+using Oceananigans.Fields: ZeroField, ConstantField
+
+==(::ZeroField, ::ZeroField) = true
+
+==(zf::ZeroField, cf::ConstantField) = 0 == cf.constant
+==(cf::ConstantField, zf::ZeroField) = ==(zf, cf)
+==(c1::ConstantField, c2::ConstantField) = c1.constant == c2.constant
+
++(a::ZeroField, b::AbstractField) = b
++(a::AbstractField, b::ZeroField) = +(b, a)
```

Comment:
It might be clearer to be a bit more explicit by writing

```julia
+(a::AbstractField, b::ZeroField) = a
```

and same for the others (I was confused when I read this code initially --- though I do understand the motivation for the other forms)

Reply:
Agreed. Fixed